### PR TITLE
Auth0 type added for `createGuardianEnrollmentTicket` as per advice on support ticket # 00750965

### PR DESCRIPTION
**Pull request by AkberIqbal**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [] Provide a URL to documentation or source code which provides context for the suggested changes: [createGuardianEnrollmentTicket](https://auth0.github.io/node-auth0/ManagementClient.html#createGuardianEnrollmentTicket)
- [N/A] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

This change was advised by A Straube (Developer Support Engineer II - Auth0) via email when I raised a support ticket (ref # 00750965)

### CODE REVIEW

File: types/auth0/index.d.ts [hunk 1]
```diff
@@ -2007,6 +2054,12 @@ export class ManagementClient<A = AppMetadata, U = UserMetadata> {
     deleteGuardianEnrollment(params: ObjectWithId): Promise<void>;
     deleteGuardianEnrollment(params: ObjectWithId, cb?: (err: Error) => void): void;
 
+    createGuardianEnrollmentTicket(data: sendEnrollmentTicketData): Promise<sendEnrollmentTicketResponse>;
+    createGuardianEnrollmentTicket(
+        data: sendEnrollmentTicketData,
+        cb?: (err: Error, data: sendEnrollmentTicketResponse) => void,
```
Comment (peterblazejewicz) [inline types/auth0/index.d.ts:2060]:
```suggestion
        cb: (err: Error, data: sendEnrollmentTicketResponse) => void,
```
this probably should not be optional

File: types/auth0/index.d.ts [hunk 2]
```diff
@@ -1521,16 +1544,32 @@ export interface DeviceCredential {
     last_used?: string;
 }
 
+export interface sendEnrollmentTicketData {
```
Comment (peterblazejewicz) [inline types/auth0/index.d.ts:1547]:
```suggestion
export interface SendEnrollmentTicketData {
```
IMO this is not widely used notation for JavaScript/TS interfaces

File: types/auth0/index.d.ts [hunk 3]
```diff
@@ -1521,16 +1544,32 @@ export interface DeviceCredential {
...
+    user_id: string;
+    send_mail?: boolean;
+}
+
+export interface sendEnrollmentTicketResponse {
```
Comment (peterblazejewicz) [inline types/auth0/index.d.ts:1552]:
```suggestion
export interface sendEnrollmentTicketResponse {
```
ditto about CamelCase notation for interfaces

File: types/auth0/index.d.ts (hunk 1 above)
Comment (AkberIqbal) [reply types/auth0/index.d.ts]: Thank you Peter, changes made

**Comment by AkberIqbal** [comment]
Hi @peterblazejewicz,
changes made as advised, please help review, 
Thanks

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@AkberIqbal thanks!

**Comment by AkberIqbal** [comment]
Ready to merge

**Pull request merged**